egs/egs_ref.py

```diff
@@ -117,18 +117,38 @@ def __init__(self):
         self.txpool_block = pd.DataFrame() # txp data at block
         self.txpool_by_gp = pd.DataFrame() # txp data grouped by gp
         self.got_txpool = False
+        self.pending_method = 'geth' # pending tx acquisition method ('geth' or 'parity')
+
+    def _get_pending_tx_hashes(self, try_methods=True):
+        """gets pending transaction hashes and autodetects method for doing so"""
+        try:
+            hashlist = []
+            if self.pending_method == 'geth':
+                txpoolcontent = web3.txpool.content
+                txpoolpending = txpoolcontent['pending']
+                for tx_sequence in txpoolpending.values():
+                    for tx_obj in tx_sequence.values():
+                        hashlist.append(tx_obj['hash'])
+                return hashlist
+            elif self.pending_method == 'parity':
+                txpoolpending = web3.manager.request_blocking('parity_pendingTransactions',[])
+                for tx_obj in txpoolpending:
+                    hashlist.append(tx_obj['hash'])
+                return hashlist
+        except ValueError as ve:
+            if ve.args[0]['code']==-32601: # method not found
```

Comment:
I'd prefer to see the expected node type abstracted to a command-line flag or a setting in the configuration file, vs. doing this eat-the-exception-and-guess-again method.